plugins/perspective_api/newhelm/annotators/perspective_api.py

```diff
@@ -85,28 +85,46 @@ def annotate_test_item(
                 self.client = self._load_client()
 
         requests = []
+        empty_completion_indexes = set()
+        request_index = 0
         for interaction in interactions:
             for completion in interaction.response.completions:
-                requests.append(
-                    self.client.comments().analyze(
-                        body=self._make_analyze_comment_request(completion.text)
+                if completion.text == "":
+                    empty_completion_indexes.add(request_index)
+                else:
+                    requests.append(
+                        self.client.comments().analyze(
+                            body=self._make_analyze_comment_request(completion.text)
+                        )
                     )
-                )
+                request_index += 1
+
         with self._client_lock:
             responses = _batch_execute_requests(
```

Comment:
Can you add an early return to `_batch_execute_requests` in the event that `requests` is empty? I'm not sure what that endpoint will do with no requests, and it seems like we can save the hassle of establishing a connection.

This will protect us in situations where the only completion is an empty string.

Reply:
Seems easy and makes sense to me. Thanks. Done.

Reply:
Minor, but can you put the return into _batch_execute_requests instead of the wrapper. That way if we reuse that function it will also get the early return logic.

Reply:
Done.